File: android/lib/src/main/java/me/anasmusa/learncast/lib/component/SheetMenuButton.kt
package me.anasmusa.learncast.lib.component

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

@Composable
fun SheetMenuButton(
    modifier: Modifier = Modifier,
    icon: ImageVector?,
    title: String,
    clip: Boolean = true,
    horizontalArrangement: Arrangement.Horizontal = Arrangement.Start,
    padding: PaddingValues = PaddingValues(12.dp),
    paddingBetween: Dp = 8.dp,
    onClick: () -> Unit,
) {
    PrimaryButton(
        modifier =
            modifier
                .fillMaxWidth()
                .height(56.dp),
        icon = icon,
        title = title,
        clip = clip,
        padding = padding,
        paddingBetween = paddingBetween,
        horizontalArrangement = horizontalArrangement,
        backgroundColor = MaterialTheme.colorScheme.surfaceContainerLowest,
        onClick = onClick,
    )
}

@Composable
fun SheetMenuWhiteButton(
    modifier: Modifier = Modifier,
    icon: ImageVector?,
    title: String,
    clip: Boolean = true,
    horizontalArrangement: Arrangement.Horizontal = Arrangement.Start,
    padding: PaddingValues = PaddingValues(12.dp),
    paddingBetween: Dp = 8.dp,
    onClick: () -> Unit,
) {
    PrimaryButton(
        modifier =
            modifier
                .fillMaxWidth()
                .height(56.dp),
        icon = icon,
        title = title,
        clip = clip,
        padding = padding,
        paddingBetween = paddingBetween,
        horizontalArrangement = horizontalArrangement,
        backgroundColor = MaterialTheme.colorScheme.secondaryContainer,
        onClick = onClick,
        titleColor = MaterialTheme.colorScheme.background,
    )
}

File: ios/lib/component/SheetMenuButton.swift
//
//  SheetMenuButton.swift
//  ios
//
//  Created by Anas Erkinjonov on 03/02/26.
//

import SwiftUI

struct SheetMenuButton: View {
    let icon: String?
    let title: String
    var clip: Bool = true
    var horizontalAlignment: HorizontalAlignment = .leading
    var padding: EdgeInsets = EdgeInsets(top: 12, leading: 12, bottom: 12, trailing: 12)
    var spacing: CGFloat = 8
    let onClick: () -> Void

    var body: some View {
        PrimaryButton(
            title: title,
            icon: icon,
            clip: clip,
            padding: padding,
            spacing: spacing,
            backgroundColor: Colors.surfaceContainerLowest,
            horizontalAlignment: horizontalAlignment,
            height: 56,
            onClick: onClick
        )
    }
}

struct SheetMenuWhiteButton: View {
    let icon: String?
    let title: String
    var clip: Bool = true
    var horizontalAlignment: HorizontalAlignment = .leading
    var padding: EdgeInsets = EdgeInsets(top: 12, leading: 12, bottom: 12, trailing: 12)
    var spacing: CGFloat = 8
    let onClick: () -> Void

    var body: some View {
        PrimaryButton(
            title: title,
            icon: icon,
            clip: clip,
            padding: padding,
            spacing: spacing,
            titleColor: Colors.onSecondaryContainer,
            backgroundColor: Colors.secondaryContainer,
            horizontalAlignment: horizontalAlignment,
            height: 56,
            onClick: onClick
        )
    }
}
